pyerrors/fits.py

```diff
@@ -151,6 +151,12 @@ def func_b(a, x):
         For details about how the covariance matrix is estimated see `pyerrors.obs.covariance`.
         In practice the correlation matrix is Cholesky decomposed and inverted (instead of the covariance matrix).
         This procedure should be numerically more stable as the correlation matrix is typically better conditioned (Jacobi preconditioning).
+    corr_matrix: array, optional
```

Comment:
I'd have the general question if we want to allow to pass the correlation matrix or the covariance matrix. The latter would allow to work with modified weights for single data points without having to mess around with the uncertainties. Imagine the situation, where you want to give more weight to points close to the continuum in an extrapolation. Together with the expected chi^2, this can have a well-defined meaning.

Reply:
Concerning the expectec chisquare: This part of the code has not yet been touched. If we want people to allow to pass any kind of covariance/correlation matrix, it has to be clear if this is meant to be a better estimator than the internal one or if it is just some modified version for use in the chisquare function. The computation and interpretation of the expected chisquare would depend on this. This requires some thought.

Reply:
(1) I think it's nice to be able to just hand over the correlation matrix because the y errors are in the code already so it's nice not to have to construct them again outside of the `least_squares` function. I also thought it would be a safer option where the `least_squares` function does most of the work for you (inverting and including the errors).
(2) Additionally I thought it would be nice to have one less restrictive option as to also incorporate modified weights for single data points like you say. (more on that below)

I suppose it would be useful to be able to hand over a covariance/correlation matrix for the expected chisquare.
Sometimes the matrix constructed in the code is ill-defined and you do not get a reliable expected chisquare. And it would also be interesting to see the effect of a covariance/correlation matrix of your choosing especially if it's to be compared with the correlated fit version with that same matrix.